app/controllers/admin/rebate_forms_controller.rb

```diff
@@ -1,9 +1,42 @@
+require 'rqrcode'
+require 'base64'
+require 'jwt'
+
 # frozen_string_literal: true
 
 class Admin::RebateFormsController < Admin::BaseController
   before_action :set_rebate_form, only: %i[show update destroy edit]
   respond_to :html, :pdf, :csv, :json
 
+  def generateqr
+    @rebate_form = RebateForm.find(params[:rebate_form_id])
+    authorize @rebate_form
+
+    # create a new token that only allows you to fetch a restricted set of details for this application only
+    # and use the same token to submit the signatures back
+    # this token is valid for 30 minutes
+    # TODO this secret should be stored in the ENV
+    # TODO the expiration time should be stored in the ENV
+    hmac_secret = 'DgD3NMLjoEsgbT7bOmpgI0svg18BMHuy0VpE6cX9xMMxznWhvEKIggR61nE/E+MOcoRi6adWUQXjiZ8/gC95ow=='
```

Comment:
secret needs to be regenerated and then kept in .env file (with example in env-example)

If this was still deploying to Heroku, you'd add to app.json, so it can be prompted for at install (or retrieved from a secret store). Not sure what the equivalent is on our new pass.